src/components/BottomBarWithIcons/index.js

```diff
@@ -23,193 +22,156 @@ import {
+    groceriesBasketData.length === 0 &&
+    productsBasketData.length === 0 &&
+    dishesBasketData.length === 0 &&
+    savedListBasketData.length === 0
 
-  if (
-    useBasketGroceres.isError ||
-    useBasketProducts.isError ||
-    useBasketDishes.isError ||
-    useBasketSavedLists.isError
-  ) {
-    return 'Fetching date error...'
-  } else if (
-    useBasketGroceres.isLoading ||
-    useBasketProducts.isLoading ||
-    useBasketDishes.isLoading ||
-    useBasketSavedLists.isLoading
-  ) {
-    return 'Loading date...'
-  } else if (
-    useBasketGroceres.isSuccess &&
-    useBasketProducts.isSuccess &&
-    useBasketDishes.isSuccess &&
-    useBasketSavedLists.isSuccess
-  ) {
-    const basketEmpty =
-      useBasketGroceres.data.length === 0 &&
-      useBasketProducts.data.length === 0 &&
-      useBasketDishes.data.length === 0 &&
-      useBasketSavedLists.data.length === 0
-
-    const handleMouseDownIcon = () => {
-      const img = document.querySelector('#returnIcon')
+  const handleMouseDownIcon = () => {
+    const img = document.querySelector('#returnIcon')
+
+    if (!!img) {
       img.setAttribute('src', return_coloredImg)
     }
+  }
+
+  const handleMouseUpIcon = () => {
+    const img = document.querySelector('#returnIcon')
 
-    const handleMouseUpIcon = () => {
-      const img = document.querySelector('#returnIcon')
+    if (!!img) {
       img.setAttribute('src', return_blackImg)
     }
+  }
 
-    const resetIcon = (
-      <Styles.IconElement>
-        <Styles.Img
-          id="returnIcon"
-          src={return_blackImg}
-          onMouseDown={() => handleMouseDownIcon()}
-          onMouseUp={() => handleMouseUpIcon()}
-          onClick={() => {
-            history.goBack()
-            dispatch(resetEditIcon())
-          }}
-        ></Styles.Img>
-      </Styles.IconElement>
-    )
+  const resetIcon = (
+    <Styles.IconElement>
+      <Styles.Img
+        id="returnIcon"
+        src={return_blackImg}
+        onMouseDown={() => handleMouseDownIcon()}
+        onMouseUp={() => handleMouseUpIcon()}
+        onClick={() => {
+          history.goBack()
+          dispatch(resetEditIcon())
+        }}
+      ></Styles.Img>
+    </Styles.IconElement>
+  )
 
-    const IconSwitch = (icon) => {
-      switch (icon) {
-        case 'save':
-          if (BasketHistory === false) {
-            return save_blackImg
-          } else {
-            return save_coloredImg
-          }
-        case 'edit':
-          if (editIcon === false) {
-            return list_coloredImg
-          } else {
-            return edit_coloredImg
-          }
-        case 'add':
-          if (addIcon === false) {
-            return add_blackImg
-          } else {
-            return add_coloredImg
-          }
-        case 'remove':
-          if (removeIcon === false) {
-            return remove_blackImg
-          } else {
-            return remove_coloredImg
-          }
+  const IconSwitch = (icon) => {
```

Comment:
to jest do przepisania na obiekt, tak jak na callu mówiliśmy

icons = {
save: {
active: save_blackImg
inactive: save_coloredImg
},
edit: {
...
}
...
}

Potem uzywasz icons[icon][BasketHistory ? 'active' : 'inactive']

I jakby co, nigdy nie pisz `x === false`, tylko `!x` w sprawdzaniu warunkow

Reply:
zdjęcia generowane są wewnątrz metody map więc musi być to funkcja zwracająca zdjęcie. Użyłem funkcji zwracającej literal object 

const IconSwitchObject = (name) => {
    const icons = {
      save: basketHistory ? save_coloredImg : save_blackImg,
      edit: editIcon ? edit_coloredImg : list_coloredImg,
      add: addIcon ? add_coloredImg : add_blackImg,
      remove: removeIcon ? remove_coloredImg : remove_blackImg,
    }

    return icons[name]
  }